tools/sigclient/pkg/ingest/ingest.go

```diff
@@ -173,18 +224,24 @@ func runIngestion(iType IngestType, rdr utils.Generator, wg *sync.WaitGroup, url
 
 	i := 0
 	var bb *bytebufferpool.ByteBuffer
+	var payload []byte
+	var err error
 	maxDuration := 2 * time.Hour
-	for continous || eventCounter < totalEvents {
+	for continuous || eventCounter < totalEvents {
 
 		recsInBatch := batchSize
-		if !continous && eventCounter+batchSize > totalEvents {
+		if !continuous && eventCounter+batchSize > totalEvents {
 			recsInBatch = totalEvents - eventCounter
 		}
 		i++
 		if iType == ESBulk {
 			bb = bytebufferpool.Get()
 		}
-		payload, err := generateBody(iType, recsInBatch, i, rdr, actLines, bb)
+		if iType == OpenTSDB {
+			payload, err = generateBodyFromPredefinedSeries(recsInBatch)
```

Comment:
you need to keep track of which series, you have gone through in the previous iteration, else this for loop is only going to send data for the first x series